src/styles/calendar.scss

```diff
@@ -0,0 +1,47 @@
+.calendar {
+  width: $squeare;
+
+  &__field {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $gap;
+    padding: 10px;
+
+    @each $weekday, $number in $weekset {
+      &--start-day--#{$weekday} :first-child {
+        margin-left: #{($width + ($border * 2) + $gap) * $number};
+      }
+    }
+
+    @for $length from 28 to 31 {
+      &--month-length--#{$length} :nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__cell {
+    display: flex;
+    width: $width;
+    height: $height;
+    border: $border solid black;
+    background-color: #eee;
+    position: relative;
+    transition-duration: 0.2s;
```

Comment:
![image](https://user-images.githubusercontent.com/98764641/199509520-766a2373-ca58-49f4-a33e-f63ac4969d0e.png)
and list all transition properties, it is better for performance